controllers/controllers/clusters/vsphere.go

```diff
@@ -2,19 +2,153 @@ package clusters
+func (v *VSphereReconciler) VsphereCredentials(ctx context.Context) (*apiv1.Secret, error) {
+	secret := &apiv1.Secret{}
+	secretKey := client.ObjectKey{
+		Namespace: "eksa-system",
+		Name:      vsphere.CredentialsObjectName,
+	}
+	if err := v.Client.Get(ctx, secretKey, secret); err != nil {
+		return nil, err
+	}
+	return secret, nil
+}
+
+func (v *VSphereReconciler) SetupEnvsAndDefaults(ctx context.Context, vsphereDatacenter *anywherev1.VSphereDatacenterConfig) error {
+	secret, err := v.VsphereCredentials(ctx)
+	if err != nil {
+		return fmt.Errorf("failed getting vsphere credentials secret: %v", err)
+	}
+
+	vsphereUsername := secret.Data["username"]
+	vspherePassword := secret.Data["password"]
+
+	if err := os.Setenv(vsphere.EksavSphereUsernameKey, string(vsphereUsername)); err != nil {
+		return fmt.Errorf("failed setting env %s: %v", vsphere.EksavSphereUsernameKey, err)
+	}
+
+	if err := os.Setenv(vsphere.EksavSpherePasswordKey, string(vspherePassword)); err != nil {
+		return fmt.Errorf("failed setting env %s: %v", vsphere.EksavSpherePasswordKey, err)
+	}
+
+	if err := vsphere.SetupEnvVars(vsphereDatacenter); err != nil {
+		return fmt.Errorf("failed setting env vars: %v", err)
+	}
+
+	if err := v.Defaulter.SetDefaultsForDatacenterConfig(ctx, vsphereDatacenter); err != nil {
+		return fmt.Errorf("failed setting default values for vsphere datacenter config: %v", err)
+	}
+
+	return nil
+}
+
+func (v *VSphereClusterReconciler) bundles(ctx context.Context, name, namespace string) (*releasev1alpha1.Bundles, error) {
+	clusterBundle := &releasev1alpha1.Bundles{}
+	err := v.capiResourceFetcher.FetchObjectByName(ctx, name, namespace, clusterBundle)
+	if err != nil {
+		return nil, err
+	}
+	return clusterBundle, nil
+}
+
+func (v *VSphereClusterReconciler) FetchAppliedSpec(ctx context.Context, cs *anywherev1.Cluster) (*c.Spec, error) {
+	return c.BuildSpecForCluster(ctx, cs, v.bundles, nil)
+}
+
+func (v *VSphereClusterReconciler) Reconcile(ctx context.Context, cluster *anywherev1.Cluster) (reconciler.Result, error) {
+	dataCenterConfig := &anywherev1.VSphereDatacenterConfig{}
+	dataCenterName := types.NamespacedName{Namespace: cluster.Namespace, Name: cluster.Spec.DatacenterRef.Name}
+	if err := v.Client.Get(ctx, dataCenterName, dataCenterConfig); err != nil {
+		return reconciler.Result{}, err
+	}
+	// Set up envs for executing Govc cmd and default values for datacenter config
+	if err := v.SetupEnvsAndDefaults(ctx, dataCenterConfig); err != nil {
+		v.Log.Error(err, "Failed to set up env vars and default values for VsphereDatacenterConfig")
+		return reconciler.Result{}, err
+	}
+
+	machineConfigMap := map[string]*anywherev1.VSphereMachineConfig{}
+
+	if cluster.Spec.ExternalEtcdConfiguration != nil {
+		mc := &anywherev1.VSphereMachineConfig{}
+		mcName := types.NamespacedName{Namespace: cluster.Namespace, Name: cluster.Spec.ExternalEtcdConfiguration.MachineGroupRef.Name}
+		if err := v.Client.Get(ctx, mcName, mc); err != nil {
```

Comment:
I see some of the objects are fetched by method defined in fetcher.go. Some (like this one) are directly calling the `Client.Get`. would be nice if we can stay consistent and use one way over another

Reply:
the resource fetcher has been removed. The current code is using Client.Get